Update swagger.yaml

nataTerr opened the pull request:
Closes #8 
Closes #10

nataTerr committed:
Update swagger.yaml

zimin-krsk commented on swagger.yaml:
```diff
@@ -1,718 +1,650 @@
+ # license:
+   # name: Apache 2.0
+   # url: http://www.apache.org/licenses/LICENSE-2.0.html
+host: api
```
хост это адрес сервера, на котором развернуто твое приложение и которое отвечает на вызовы описанных в спецификации методах.\
но т.к. ты сейчас приложение нигде не разворачиваешь, то и хоста нет

поэтому можно тоже закомментировать

zimin-krsk commented on swagger.yaml:
```diff
@@ -1,718 +1,650 @@
+   # name: Apache 2.0
+   # url: http://www.apache.org/licenses/LICENSE-2.0.html
+host: api
+basePath: /v1
```
а вот basePath как раз будет `/api/v1`

zimin-krsk commented on swagger.yaml:
```diff
@@ -1,718 +1,650 @@
-    description: "Find out more about our store"
-    url: "http://swagger.io"
+  - name: site
+    description: Online Storefront service
```
`Online Store front API service` скорее

zimin-krsk commented on swagger.yaml:
```diff
@@ -1,718 +1,650 @@
+    externalDocs:
+      description: Find out more
+      url: https://github.com/nataTerr/online-store-portal
+  - name: user
```
`accounting system`, например

nataTerr replied:
Тут не поняла, убрать ссылку на гитхаб?

zimin-krsk replied:
не, это коммент к строке
> - name: user

типа название раздела неподходящее

zimin-krsk commented on swagger.yaml:
```diff
@@ -1,718 +1,650 @@
+              $ref: '#/definitions/Catalog'
+        '400':
+          description: Invalid status value
+  /catalog/{categoryId}/parentCategories:
```
это для хлебных крошек ведь?

zimin-krsk replied:
я чет подзабыл

nataTerr replied:
Да, я это уже поправила

zimin-krsk commented on swagger.yaml:
```diff
@@ -1,718 +1,650 @@
-          description: "Pet not found"
-      security:
-      - api_key: []
+            type: array
```
чет как будто тут отступы поехали немного

nataTerr replied:
Это вообще не к делу ))) я убрала лишнее

zimin-krsk commented on swagger.yaml:
```diff
@@ -1,718 +1,650 @@
+             $ref: '#/definitions/Category'
+        '400':
+          description: Invalid ID supplied
+        '404':
```
обычно, когда возвращается коллекция объектов, то предпочтительней вернуть пустую коллекцию, когда ничего не найдено, нежели бросать ошибку\
в таком случае на фронте ничего не отобразят и не придется обрабатывать ошибку.\
в общем когда возвращаешь список, проще дать пустой ответ, типа нет ничего

zimin-krsk commented:
кстати, все лишние методы можешь просто вынести в отдельный файл, чтобы туда подглядывать\
и в твоих методах мешаться не будет

nataTerr committed:
Update swagger.yaml

nataTerr committed:
Correct responses

Replace 400 by 204

zimin-krsk commented on swagger.yaml:
```diff
@@ -1,718 +1,254 @@
-        200:
-          description: "successful operation"
+              $ref: '#/definitions/Category'
+        '204':
```
такого ответа не будет, может быть или 200 или 204\
ну т.е. технически может быть и так и так, но это запутывает логику.\
- 200 - возвращает данные, которые попросили (если массив, то пустой массив, когда ничего не найдено)
- 204 - обычно возвращается в стиле "я сделал то, что меня попросили"

zimin-krsk replied:
204 обычно на методы post и put возвращаются

nataTerr replied:
Исправила

zimin-krsk commented on swagger.yaml:
```diff
@@ -1,718 +1,254 @@
+            type: array
+            items:
+             $ref: '#/definitions/Category'
+        '204':
```
аналогично про 200 и 204

nataTerr replied:
Исправила

zimin-krsk commented on swagger.yaml:
```diff
@@ -1,718 +1,254 @@
-        type: "string"
-  Category:
-    type: "object"
+  ProductList:
```
Лучше назвать ProductPreview или ProductShortCard\
название ProductList подразумевает, что содержимое будет массивом, но это не так. Что вводит в заблуждение

nataTerr replied:
Исправила на ProductShortCard

zimin-krsk commented on swagger.yaml:
```diff
@@ -1,718 +1,254 @@
-    type: "object"
+  ProductList:
+    type: object
     properties:
```
нужно тоже указать какие поля будут обязательными, а какие нет

nataTerr replied:
Исправила

zimin-krsk commented on swagger.yaml:
```diff
@@ -1,718 +1,254 @@
+            type: array
+            items:
+              $ref: '#/definitions/ProductList'
+        '204':
```
аналогично про 200 и 204

nataTerr replied:
исправила

zimin-krsk commented on swagger.yaml:
```diff
@@ -1,718 +1,254 @@
+      pagination:
+        type: object
+        $ref: '#/definitions/Pagination'
     xml:
```
вот эта штука тебе уже не нужна, получается, потому что ты в своих методах указала, что только `application/json` формат будешь возвращать.
этот тег нужен, чтобы управлять ответом в xml формате: https://swagger.io/docs/specification/data-models/representing-xml/

nataTerr replied:
убрала везде, где встречается

zimin-krsk commented on swagger.yaml:
```diff
@@ -1,718 +1,254 @@
-          wrapped: true
+        type: string
+      price:
+        type: integer
```
вот тут интересно получается, для денежных единиц надо использовать тип number, но вот подходящего формата не существует почему-то :)
при этом [тут](https://stackoverflow.com/questions/46350701/what-is-the-type-for-currency) рекомендуют указывать `decimal`, это правильно, но спецификация это не умеет обрабатывать на этапе генерации.
Предлагаю сейчас указать
```
type: number
format: decimal
```
а потом посмотрим, чет пока ничего лучше не нашел

zimin-krsk replied:
если decimal не получится, тогда давай пока string сделаем

nataTerr replied:
Поставила string, но жду уточнения от тебя

zimin-krsk replied:
Посмотрел, у нас везде используется number. Так что пусть number будет :)

zimin-krsk commented on swagger.yaml:
```diff
@@ -1,718 +1,254 @@
+      price:
+        type: integer
+        format: int32
+      image:
```
во множественном числе надо, потому что массив

nataTerr replied:
исправила на images

zimin-krsk commented on swagger.yaml:
```diff
@@ -1,718 +1,254 @@
+        type: string
+    xml:
+      name: Category
+  Properties:
```
Лучше ProductProperties тут, а то слишком абстрактно

nataTerr replied:
изменила

nataTerr committed:
correction of mistakes

nataTerr committed:
Correct yaml

Убрала комментарии, исправила у price формат на "number", установила обязательные поля для ProductCard
